ruby/lib/ci/queue/redis/build_record.rb

```diff
@@ -48,10 +48,25 @@ def record_error(id, payload, stats: nil)
         end
 
         def record_success(id, stats: nil)
-          redis.pipelined do |pipeline|
+          deleted_count, _ = redis.pipelined do |pipeline|
             pipeline.hdel(key('error-reports'), id.dup.force_encoding(Encoding::BINARY))
             record_stats(stats, pipeline: pipeline)
           end
+          record_flaky(id) if deleted_count == 1
+          nil
+        end
+
+        def record_flaky(id, stats: nil)
+          redis.pipelined do |pipeline|
+            pipeline.sadd(
+              key('flaky-reports'),
+              id.dup.force_encoding(Encoding::BINARY)
+            )
+            pipeline.expire(key('flaky-reports'), config.redis_ttl)
+          end
+          nil
+        rescue => e
+          puts "Error recording flaky: #{e.message}"
```

Comment:
Not a fan of this `puts`, as this class isn't supposed to do any UI stuff.

Why do we even expect this to fail?